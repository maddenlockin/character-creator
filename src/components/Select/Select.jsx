import React from 'react'

export default function Select({
    top, 
    middle, 
    bottom, 
    setTop, 
    setMiddle, 
    setBottom, 
    judgement, 
    setJudgement, 
    handleClick
}) {
    const tOptions = ['tophat', 'crown', 'church', 'bunny-ear']
    const midOptions = ['chillin', 'furry', 'squish', 'wrinkles']
    const botOptions = ['cat', 'green-dragon', 'pizza', 'pink-dragon']
    return (
        <div className='select'>
            <label>
                Top
                <select value={top} onChange={(e) => setTop(e.target.value)}>
                    {tOptions.map((option) => {
                    return <option key={option}>{option}</option>;
                })}
                </select>
            </label>

            <label>
                Middle
                <select value={middle} onChange={(e) => setMiddle(e.target.value)}>
                    {midOptions.map((option) => {
                    return <option key={option}>{option}</option>;
                })}
                </select>
            </label>

            <label>
                Bottom
                <select value={bottom} onChange={(e) => setBottom(e.target.value)}>
                    {botOptions.map((option) => {
                    return <option key={option}>{option}</option>;
                })}
                </select>
            </label>

            <label>
                Judgements
                <input
                    aria-label='judgement input'
                    type='text'
                    value={judgement}
                    onChange={(e) => setJudgement(e.target.value)}
                />
            </label>

            <button type='button' onClick={handleClick}>
                Judge
            </button>
        </div>
    )
}
