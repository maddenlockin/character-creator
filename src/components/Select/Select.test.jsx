import { render, screen } from "@testing-library/react";
import Select from "./Select";

it("should render select component", () => {
    const { container } = render(<Select />);
    const selectTop = screen.getByLabelText(/Top/);
    const selectMiddle = screen.getByLabelText(/Middle/);
    const selectBottom = screen.getByLabelText(/Bottom/);
    const input = screen.getByLabelText(/Judgements/);

    expect(selectTop).toBeInTheDocument();
    expect(selectMiddle).toBeInTheDocument();
    expect(selectBottom).toBeInTheDocument();
    expect(input).toBeInTheDocument();
    expect(container).toMatchSnapshot();
});
