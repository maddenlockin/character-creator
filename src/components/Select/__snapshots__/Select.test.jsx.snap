// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render select component 1`] = `
<div>
  <section
    class="select"
  >
    <label>
      Top
      <select>
        <option>
          tophat
        </option>
        <option>
          crown
        </option>
        <option>
          church
        </option>
        <option>
          bunny-ear
        </option>
      </select>
    </label>
    <label>
      Middle
      <select>
        <option>
          chillin
        </option>
        <option>
          furry
        </option>
        <option>
          squish
        </option>
        <option>
          wrinkles
        </option>
      </select>
    </label>
    <label>
      Bottom
      <select>
        <option>
          cat
        </option>
        <option>
          green-dragon
        </option>
        <option>
          pizza
        </option>
        <option>
          pink-dragon
        </option>
      </select>
    </label>
    <label>
      Judgements
      <input
        aria-label="judgement input"
        type="text"
        value=""
      />
    </label>
    <button
      type="button"
    >
      Judge
    </button>
    <div>
      <label />
    </div>
  </section>
</div>
`;

exports[`should render select component 2`] = `
<div>
  <section
    class="select"
  >
    <label>
      Top
      <select>
        <option>
          tophat
        </option>
        <option>
          crown
        </option>
        <option>
          church
        </option>
        <option>
          bunny-ear
        </option>
      </select>
    </label>
    <label>
      Middle
      <select>
        <option>
          chillin
        </option>
        <option>
          furry
        </option>
        <option>
          squish
        </option>
        <option>
          wrinkles
        </option>
      </select>
    </label>
    <label>
      Bottom
      <select>
        <option>
          cat
        </option>
        <option>
          green-dragon
        </option>
        <option>
          pizza
        </option>
        <option>
          pink-dragon
        </option>
      </select>
    </label>
    <label>
      Judgements
      <input
        aria-label="judgement input"
        type="text"
        value=""
      />
    </label>
    <button
      type="button"
    >
      Judge
    </button>
    <div>
      <label />
    </div>
  </section>
</div>
`;
